### ISSUE TITLE
NLB created after the second time didn't route UDP packets in IP mode

### ISSUE THREAD

**Issue by dbaba** [kubernetes-sigs/aws-load-balancer-controller#2074]
I'm trying to use IP mode NLB for my UDP app but UDP packets didn't seem to arrive at the app container in a pod after TargetGroup showed IP target was healthy.
On the first attempt, IP mode seemed to work (meaning UDP packets were reachable to the destination pod) but after destroying and recreating NLB by `kubectl delete` and `kubectl apply`, it has never worked.
In Instance mode, NLB worked as expected. UDP packets were always reachable whenever NLB was up. So I'd say the pod definition (Deployment) looks good.

Is there something I'm missing? Any suggestions are appreciated.

I'm using Kubernetes v1.20 on EKS, aws-load-balancer-controller v2.2.0, and VPC CNI plugin v1.7.10.

Here is the YAML file I tested (But this is not working as the container image name is not provided as it's my private image). The installation/uninstallation steps I performed are also described below.

```yaml
# INSTALLATION for testing
# ===
# kubectl create namespace my-udp-app
# kubectl apply -f THIS_K8S_YAML_FILE -n my-udp-app
### send test packets
# nc -u NLB-DNS-NAME 8081
# ===
# UNINSTALLATION
# ===
# kubectl delete -f THIS_K8S_YAML_FILE -n my-udp-app
# kubectl delete namespace my-udp-app
---
apiVersion: v1
kind: ServiceAccount
metadata:
  name: my-udp-app
  labels:
    app.kubernetes.io/name: my-udp-app
---
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: my-udp-app
  labels:
    app.kubernetes.io/name: my-udp-app
roleRef:
  kind: ClusterRole
  name: cluster-admin
  apiGroup: rbac.authorization.k8s.io
subjects:
  - kind: ServiceAccount
    name: my-udp-app
    namespace: my-udp-app
---
apiVersion: apps/v1
kind: Deployment
metadata:
  name: release-my-udp-app
spec:
  replicas: 1
  selector:
    matchLabels:
      app.kubernetes.io/name: my-udp-app
  template:
    metadata:
      labels:
        app.kubernetes.io/name: my-udp-app
    spec:
      serviceAccountName: my-udp-app
      securityContext:
        {}
      containers:
        - name: my-udp-app
          securityContext:
            {}
          image: <my-udp-app image name here>
          imagePullPolicy: Always
          ports:
            - name: http
              containerPort: 8080
              protocol: TCP
            - name: udp-app-port
              containerPort: 8081
              protocol: UDP
          livenessProbe:
            httpGet:
              path: /status
              port: 8080
            initialDelaySeconds: 15
            periodSeconds: 20
          readinessProbe:
            httpGet:
              path: /status
              port: 8080
            initialDelaySeconds: 10
            periodSeconds: 10
          resources:
            {}
---
apiVersion: v1
kind: Service
metadata:
  name: release-my-udp-app
  labels:
    app.kubernetes.io/name: my-udp-app
  annotations:
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-healthy-threshold: "3"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-interval: "30"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-path: /status
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-port: "8080"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-protocol: http
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-timeout: "6"
    service.beta.kubernetes.io/aws-load-balancer-healthcheck-unhealthy-threshold: "3"
    service.beta.kubernetes.io/aws-load-balancer-nlb-target-type: ip
    service.beta.kubernetes.io/aws-load-balancer-scheme: internet-facing
    service.beta.kubernetes.io/aws-load-balancer-type: external
spec:
  type: LoadBalancer
  ports:
    - port: 8081
      protocol: UDP
      name: udp-app-port
  selector:
    app.kubernetes.io/name: my-udp-app

```

**Comment by kishorj** [comment]
VPC CNI older than v1.8.0, has an issue that prevents the pods attached to secondary ENI from receiving client source IP preserved traffic (udp traffic has client source IP preserved). CNI issue - https://github.com/aws/amazon-vpc-cni-k8s/issues/1392.

The fix is included in v1.8.0 release. For details, please refer to the VPC CNI release notes [[link](https://github.com/aws/amazon-vpc-cni-k8s/releases/tag/v1.8.0)].

**Comment by dbaba** [comment]
Thank you for the helpful info.
I'm using Helm for installing VPC CNI but the chart is not yet released. I will try v1.8.0 after the chart is coming.

**Comment by dbaba** [comment]
I close the issue as it's resolved by VPC CNI plugin v1.8.0.
I tested with [aws/eks-charts v0.0.53](https://github.com/aws/eks-charts/releases/tag/v0.0.53).
